src/DataManagement/MonthlyReport/MonthlyReportCalculator.cs

```diff
@@ -104,22 +104,34 @@ private void CalculateGoalReport()
             }
 
             Result.HasGoals = true;
-            var min = decimal.MaxValue;
-            var minGoal = new Goal();
-            var maxGoal = new Goal();
-            var max = 0M;
-            foreach (var item in Goals)
+           
+          
+
+            var maxGoal = Goals.Aggregate((expensive, next) => next.Amount >= expensive.Amount ? next : expensive);
+            var minGoal = Goals.Aggregate((cheapest, next) => (next.Amount <= cheapest.Amount && next.Amount >= AllSavings)  ? next : cheapest);
```

Comment:
So how does aggregate work? Does it just go through the list and applies a certain function as it goes?

Reply:
Main concept is that variables `expensive` and ` cheapest` are for keeping results, and next is the list item currently in iteration. And usage of ternary operator allows to assign values to these "result" variables according to the condition. And after iterating it returns what's in that "result" variable. There is  ability to modify result in the end(e.g. divide by the number of items to get average if we're adding items) but it was not needed here. At least that's how I understand it